h2o-parsers/h2o-parquet-parser/src/main/java/water/parser/parquet/ChunkConverter.java

```diff
@@ -84,22 +88,35 @@ private PrimitiveConverter newConverter(int colIdx, byte vecType, PrimitiveType
   }
 
   private static class StringConverter extends PrimitiveConverter {
-
+    // Maximum array size (Integer.MAX_VALUE - 8) minus one place for a trailing zero
     private final BufferedString _bs = new BufferedString();
     private final int _colIdx;
     private final WriterDelegate _writer;
     private final boolean _dictionarySupport;
     private String[] _dict;
+    private long _writtenAmount;
+    private boolean _overflew;
 
     StringConverter(WriterDelegate writer, int colIdx, boolean dictionarySupport) {
       _colIdx = colIdx;
       _writer = writer;
       _dictionarySupport = dictionarySupport;
+      _writtenAmount = 0;
+      _overflew = false;
     }
 
     @Override
     public void addBinary(Binary value) {
+      if(_overflew) return;
       _bs.set(StringUtils.bytesOf(value.toStringUsingUTF8()));
+      _writtenAmount += _bs.length();
+      long lenDifference = _writtenAmount - MAX_STR_LEN;
+      if(lenDifference > 0){
+        _bs.setLen(MAX_STR_LEN);
```

Comment:
Why do you set the length here? It seems wrong

Reply:
First, MAX_STR_LEN is indeed wrong. 

IF there is MAX_STR_LEN - n bytes written, it would be nice to write the remaining n at least. In the _bs BufferedString, the new Value which can be Integer.MAX_VALUE (up to 8 bytes above MAX_STR_LEN) is written at this point. 

What I want to do is to find out which portion of the buffered string fits even after overflow. Should we drop it anyway ?

Reply:
I think it is better to throw away the whole value. It seems cleaner than to leave there a broken string. At least you can say that values that were successfully loaded are all correct.